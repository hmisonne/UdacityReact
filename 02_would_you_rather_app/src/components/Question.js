import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import Card from 'react-bootstrap/Card';
import PropTypes from 'prop-types'
import {formatDate} from '../utils/helpers'

class Question extends Component {
	static propTypes = {
		question: PropTypes.object.isRequired,
		author: PropTypes.object.isRequired,
	}
	toPoll = (e, id) => {
		e.preventDefault()
		this.props.history.push(`/questions/${id}`)
	}
	render(){
		const {question, author} = this.props
		const formattedDate = formatDate(question.timestamp)
		return(
			<Card>
			  <Card.Header>{formattedDate} - {author.name} asks:</Card.Header>
			  <Card.Body>
				  <img
			          src={author.avatarURL}
			          alt={`Avatar of ${author.name}`}
			          className='avatar'
			        />
			    <Card.Title>Would you rather?</Card.Title>

			    <Card.Text>
			       {question.optionOne.text} Or {question.optionTwo.text}
			    </Card.Text>
			    <button className="btn btn-primary" onClick={(e) => this.toPoll(e, question.id)}>View Poll</button>
			  </Card.Body>
			</Card>
		)
	}
}



export default withRouter(Question)