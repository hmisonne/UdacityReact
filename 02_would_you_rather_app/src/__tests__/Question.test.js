import React from 'react';
import Question from '../components/Question';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Route } from 'react-router-dom'

const questionSample = {
    id: '8xf0y6ziyjabvozdd253nd',
    author: 'sarahedo',
    timestamp: 1467166872634,
    optionOne: {
      votes: ['sarahedo'],
      text: 'have horrible short term memory',
    },
    optionTwo: {
      votes: [],
      text: 'have horrible long term memory'
    }
  }

const authorSample = {
    id: 'sarahedo',
    name: 'Sarah Edo',
    avatarURL: "https://tylermcginnis.com/would-you-rather/sarah.jpg",
    answers: {
      "8xf0y6ziyjabvozdd253nd": 'optionOne',
      "6ni6ok3ym7mf1p33lnez": 'optionTwo',
      "am8ehyc8byjqgar0jgpub9": 'optionTwo',
      "loxhs1bqm25b708cmbf3g": 'optionTwo'
    },
    questions: ['8xf0y6ziyjabvozdd253nd', 'am8ehyc8byjqgar0jgpub9']
  }

test('Question component renders correctly', () => {
  const component = renderer.create(
    <Router>
      <Question 
          question = {questionSample}
          author = {authorSample}
          />
    </Router>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});