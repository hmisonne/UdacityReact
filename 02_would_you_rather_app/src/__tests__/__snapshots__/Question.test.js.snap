// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    7:21:PM | 6/28/2016
     - 
    Sarah Edo
     asks:
  </div>
  <div
    className="card-body"
  >
    <img
      alt="Avatar of Sarah Edo"
      className="avatar"
      src="https://tylermcginnis.com/would-you-rather/sarah.jpg"
    />
    <div
      className="card-title h5"
    >
      Would you rather?
    </div>
    <p
      className="card-text"
    >
      have horrible short term memory
       Or 
      have horrible long term memory
    </p>
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      View Poll
    </button>
  </div>
</div>
`;
